src/idl2json/src/typed_conversion.rs

```diff
@@ -16,12 +21,25 @@ use crate::{
 /// - Fields are never added, even if the schema suggests that some fields are missing.
 ///
 /// The data is preserved at all cost, the schema is applied only to make the data easier to understand and use.
+///
+/// Note: The textual format in parentheses `(  )` represents IDLArgs containing
+/// zero or more IDLValues.  Unless you definitely wish to convert a single value
+/// you may wish to consider `idl_args2json_with_weak_names` instead.
 pub fn idl2json_with_weak_names(
     idl: &IDLValue,
     idl_type: &IDLType,
     options: &Idl2JsonOptions,
 ) -> JsonValue {
     match (idl, idl_type) {
+        (idl, IDLType::VarT(type_name)) => {
+            if let Some(resolved_type) = get_type_from_any(&options.prog, type_name) {
```

Comment:
A silly question, are type_name duplication possible/allowed?

Reply:
Discussed.

Reply:
Done: I have updated the comment on the `prog` field in the arguments with contextual information, typical usage and a warning about conflicting type definitions.